runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPanel.java

```diff
@@ -529,6 +532,34 @@ public void focusLost(FocusEvent e)
 				item.add(button, BorderLayout.EAST);
 			}
 
+			if (cid.getType() == File.class)
+			{
+				JFileChooser fileChooser = new JFileChooser();
+				fileChooser.setDialogTitle("Select a custom cursor image");
+				fileChooser.setAcceptAllFileFilterUsed(false);
+				FileNameExtensionFilter filter = new FileNameExtensionFilter("image files", "png", "gif", "jpeg", "jpg", "bmp", "ico", "cur");
+				fileChooser.addChoosableFileFilter(filter);
+
+				JButton chooserButton = new JButton("Select Image File");
+				chooserButton.setPreferredSize(new Dimension(140, 20));
+				chooserButton.addActionListener(e ->
+				{
+					int returnValue = fileChooser.showOpenDialog(null);
+					if (returnValue == JFileChooser.APPROVE_OPTION)
+					{
+						changeConfiguration(listItem, config, fileChooser, cd, cid);
+						chooserButton.setText(fileChooser.getSelectedFile().getName());
+					}
+				});
+				String currentFilePath = configManager.getConfiguration(cd.getGroup().value(), cid.getItem().keyName());
+				String fileName = currentFilePath.substring(currentFilePath.lastIndexOf("\\") + 1);
```

Comment:
display full path is probably more usable for other configs

Reply:
I feel like I don't have much space in the sidebar. How would I render the full path?